starlite/routes.py

```diff
@@ -304,6 +306,86 @@ async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
             await fn(**parsed_kwargs)
 
 
+class WSMessageRoute(BaseRoute):
+    __slots__ = (
+        "route_handler",
+        "handler_parameter_model",
+        # the rest of __slots__ are defined in BaseRoute and should not be duplicated
+        # see: https://stackoverflow.com/questions/472000/usage-of-slots
+    )
+
+    @validate_arguments(config={"arbitrary_types_allowed": True})
+    def __init__(
+        self,
+        *,
+        path: str,
+        route_handler: WSMessageHandler,
+    ):
+        self.route_handler = route_handler
+        self.handler_parameter_model: Optional[KwargsModel] = None
+        super().__init__(
+            path=path,
+            scope_type=ScopeType.WEBSOCKET,
+            handler_names=[get_name(cast(AnyCallable, route_handler.fn))],
+        )
+
+    async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
+        """
+        Holds the websocket connection and feeds parsed/validated messages into the handler.
+
+        - create WebSocket
+        - accept the connection after running guards
+        - iterate over the socket
+        - parse each message into the type of the handler's `message` kwarg
+        - invoke handler with resolved dependencies
+        """
+        assert self.handler_parameter_model, "handler parameter model not defined"
+        # `data` must be declared in the handler signature, but we manually provide the kwarg
```

Comment:
in this case you st make sure to validate this kwarg is in place

Reply:
So, look in the `Starlite` class itself, where the signature models are actually generated - this also needs to be updated.

Reply:
will check out and report back